### PR TITLE
adds tests and a fix for clean exit

### PR DESCRIPTION
**Pull request by brianleroux**
- adds tape to start testing out the api and harden the code
- adds tap-spec for nice reporting
- fixes a bug on closeRTM for clean exit

### CODE REVIEW

File: test/Slack_web_api-test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  t.ok(Botkit, 'Botkit exists')
+  t.ok(Botkit.core, 'Botkit.core exists')
+  t.ok(Botkit.slackbot, 'Botkit.slackbot exists')
+  console.log(Botkit)
+})
+
+test('can start and then stop a bot', t=> {
+
+  var controller = Botkit.slackbot({debug:false}) 
+  var bot = controller.spawn(token).startRTM((err, bot, payload)=> {
```
Comment (anonrig) [inline test/Slack_web_api-test.js]: If you're using arrow functions, why not use const or let in all of your code?

**Comment by RafaelCosman** [comment]
@brianleroux thanks for this pull request. Sorry for taking so long to look at it. I'll test it shortly.

**Comment by benbrown** [comment]
As I discussed with @brianleroux, we just need to make a small change to the way the interval is being cleared before we can merge this to avoid breaking the slack button bot functionality where multiple bots are running in one process...

**Comment by benbrown** [comment]
OK! I made the change I wanted and merged this in!

Woo, tests!!

**Comment by RafaelCosman** [comment]
Thanks for looking at this, @benbrown!

**Comment by anonrig** [comment]
If this pull request is merged, botkit will not work on Nodejs  < 4.0.0

**Comment by benbrown** [comment]
@anonrig only the tests are > 4 though right?

**Comment by benbrown** [comment]
@anonrig I just refactored the tests to be more friendly to earlier Nodes.

Let me know if you have other concerns!

**Comment by anonrig** [comment]
Yes only the tests were Node 4+ supported. Right now it seems fine. But in the future it is important to switch to NodeJS 4+. At least the LTS version 4.2.3 @benbrown. What do you think?

**Pull request merged**